lib/samlp.js

```diff
@@ -32,10 +49,54 @@ function getSamlRequest(samlRequest, callback) {
           return callback(new Error('Invalid SAML Request'));
         }
 
-        callback(null, xml);
+        if (!options.signedRequest){
+          return callback(null, xml);
+        }
+        
+        // Http-Redirect binding can have the signature in query instead of in the AuthnRequest
+        // We pass the signature and algorithm if found.
+        if (options.signature){
+          if (!options.sigAlg){
+            return callback(new Error('Signature Algorithm is missing'));
+          }
+          
+          if (!algorithmSignatures[options.sigAlg]){
+            return callback(new Error('Invalid signature algorithm. Supported algorithms are http://www.w3.org/2001/04/xmldsig-more#rsa-sha1 and http://www.w3.org/2001/04/xmldsig-more#rsa-sha256'));
+          }
+
+          if (!options.signingCert){
+            return callback(new Error('Make sure you configure the signing certificate to be able to validate the SAML Request'));
+          }
+
+          var signedParams; 
+          
+          if (options.relayState){
```

Comment:
Maybe this ? `querystring.stringify()` uses insertion order so the order in which we add the parameters is important (It changes the signature value)

```
 var signedParams =  {
      SAMLRequest: samlRequest,
}; 
          
if (options.relayState){
      signedParams.RelayState = options.relayState;
}

signedParams.SigAlg = options.sigAlg;
```